metals/src/main/scala/scala/meta/internal/metals/debug/ExtendedScalaMainClass.scala

```diff
@@ -61,10 +73,61 @@ object ExtendedScalaMainClass {
     s"$safeJavaHome $jvmOptsString -classpath \"$classpathString\" $mainClass $argumentsString"
   }
 
+  /**
+   * Create a manifest jar containing the classpath.
+   * It's created in `.metals/tmp` with the name derived
+   * from the classpath MD5 sum so that it's not created
+   * every time.
+   *
+   * @param classpath classpath to put into the jar file
+   * @param workspace current workspace path
+   * @return path to the create jar as string
+   */
+  private def tempManifestJar(
+      classpath: List[String],
+      workspace: AbsolutePath,
+  ): String = {
+
+    val classpathDigest = MD5.compute(classpath.mkString)
+    val manifestJar =
+      workspace
+        .resolve(Directories.tmp)
+        .resolve(s"classpath_${classpathDigest}.jar")
+
+    if (!manifestJar.exists) {
+
+      manifestJar.touch()
+      manifestJar.toNIO.toFile().deleteOnExit()
+
+      val classpathStr =
+        classpath
+          .map(path =>
+            URIEncoderDecoder.encode(Paths.get(path).toUri().toString())
+          )
+          .mkString(" ")
+
+      val manifest = new jar.Manifest()
+      manifest.getMainAttributes.put(
+        jar.Attributes.Name.MANIFEST_VERSION,
+        "1.0",
+      )
+      manifest.getMainAttributes.put(
+        jar.Attributes.Name.CLASS_PATH,
+        classpathStr,
+      )
+
+      val out = Files.newOutputStream(manifestJar.toNIO)
+      Using.resource(new jar.JarOutputStream(out, manifest)) { _.close() }
```

Comment:
I'm still a bit confused by this. Why do we need to call close on the output stream that we create here? Couldn't this just be `identity` and `close` will automatically be called?

Reply:
Wait, will it be automatically called? I didn't know that

Reply:
That's the whole point of `Using`, isn't it? :D

Reply:
OMG, I actually didn't know :sweat_smile:

Reply:
It's fixed btw. thanks for pointing this out!